addons/sale_timesheet/models/product.py

```diff
@@ -68,6 +68,14 @@ def _onchange_type(self):
             self.invoice_policy = 'order'
         return res
 
+    @api.onchange('service_policy')
+    def _onchange_service_policy(self):
+        if self.service_tracking != 'no' and self.service_policy == 'delivered_timesheet':
+            if self.project_id and not self.project_id.allow_timesheets:
+                self.project_id = False
+            elif self.project_template_id and not self.project_template_id.allow_timesheets:
+                self.project_template_id = False
```

Comment:
I think we should convert it into a compute method in the case where we change the `service_policy` field in a product in the code or something else. In fact, we have a `_compute_service_policy` method to compute the service_policy, I think this onchange could not be triggered in certain case.